### PR TITLE
Refactore DependencyInjection

### LINKED ISSUE #728: Refactore DependencyInjection
While working on #697, #727 that should close #559 we decided to refactor the static dependency injection container located in core into separated container for each driver. This includes 

- [x] Add service to msbuild
- [x] Add service to console
- [x] Add service to collectors
- [x] Add service to tests
- [x] remove ```DependencyInjection```from core
- [x] remove ```DependencyInjectionExtensions```

### PR DESCRIPTION
**Pull request by daveMueller**
closes #728

### CODE REVIEW

**Comment by MarcoRossignoli** [comment]
I'll take a look asap thx!

**Comment by MarcoRossignoli** [comment]
We're close to next release, after that I'll review and merge this one.

File: src/coverlet.msbuild.tasks/Services.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace coverlet.msbuild.tasks
```
Comment (MarcoRossignoli) [inline src/coverlet.msbuild.tasks/Services.cs]: nit: namespace `Coverlet.MSbuild.Tasks`

File: src/coverlet.msbuild.tasks/Services.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+{
+    internal static class Services
```
Comment (MarcoRossignoli) [inline src/coverlet.msbuild.tasks/Services.cs]: Could be more elegant create a base class
```cs
public abstract class BaseTask : Task
{
   protected static IServiceProvider ServiceProvider { get;set; }
}
...
public class InstrumentationTask : BaseTask 
...
public class CoverageResultTask : BaseTask 
...
```
And build container 
```cs
...
ServiceProvider = serviceCollection.BuildServiceProvider();
...
```

File: src/coverlet.msbuild.tasks/Services.cs (hunk 1 above)
Comment (daveMueller) [reply src/coverlet.msbuild.tasks/Services.cs]: Done

File: src/coverlet.msbuild.tasks/Services.cs (hunk 2 above)
Comment (daveMueller) [reply src/coverlet.msbuild.tasks/Services.cs]: Done

**Comment by MarcoRossignoli** [comment]
@daveMueller I needed to do a fast update on code, can you resolve conflicts?You can skip collectors I took some of your code and merged.

**Comment by daveMueller** [comment]
> @daveMueller I needed to do a fast update on code, can you resolve conflicts?You can skip collectors I took some of your code and merged.

No trouble at all. I resolved the conflicts.

File: src/coverlet.collector/DataCollection/CoverageManager.cs [hunk 3]
```diff
@@ -22,24 +22,22 @@ internal class CoverageManager
 
         public IReporter[] Reporters { get; }
 
-        public CoverageManager(CoverletSettings settings, TestPlatformEqtTrace eqtTrace, TestPlatformLogger logger, ICoverageWrapper coverageWrapper,
-                               IInstrumentationHelper instrumentationHelper, IFileSystem fileSystem, ISourceRootTranslator sourceRootTranslator)
+        public CoverageManager(CoverletSettings settings, TestPlatformEqtTrace eqtTrace, ILogger logger,
```
Comment (MarcoRossignoli) [inline src/coverlet.collector/DataCollection/CoverageManager.cs:25]: Please revert update to this file and keep master alignment.

File: src/coverlet.collector/DataCollection/CoverletCoverageCollector.cs [hunk 4]
```diff
@@ -132,7 +131,7 @@ private void OnSessionStart(object sender, SessionStartEventArgs sessionStartEve
                 _serviceProvider = _serviceCollectionFactory(_eqtTrace, _logger, coverletSettings.TestModule).BuildServiceProvider();
 
                 // Get coverage and attachment managers
-                _coverageManager = new CoverageManager(coverletSettings, _eqtTrace, _logger, _coverageWrapper,
+                _coverageManager = new CoverageManager(coverletSettings, _eqtTrace, _serviceProvider.GetRequiredService<ILogger>(), _coverageWrapper,
```
Comment (MarcoRossignoli) [inline src/coverlet.collector/DataCollection/CoverletCoverageCollector.cs:134]: Keep master version, we pass also `_eqTrace` and it's ok, it's the same object if we have access we can use it.

File: src/coverlet.console/Program.cs [hunk 5]
```diff
@@ -28,13 +28,13 @@ static int Main(string[] args)
...
 
-            DependencyInjection.Set(serviceCollection.BuildServiceProvider());
+            var serviceProvider = serviceCollection.BuildServiceProvider();
 
-            var logger = (ConsoleLogger)DependencyInjection.Current.GetService<ILogger>();
-            var fileSystem = DependencyInjection.Current.GetService<IFileSystem>();
-            var sourceTranslator = DependencyInjection.Current.GetService<ISourceRootTranslator>();
+            var logger = (ConsoleLogger) serviceProvider.GetService<ILogger>();
+            var fileSystem = serviceProvider.GetService<IFileSystem>();
+			var sourceTranslator = serviceProvider.GetService<ISourceRootTranslator>();
```
Comment (MarcoRossignoli) [inline src/coverlet.console/Program.cs:37]: nit: spaces

File: src/coverlet.collector/DataCollection/CoverageManager.cs (hunk 3 above)
Comment (daveMueller) [reply src/coverlet.collector/DataCollection/CoverageManager.cs]: Done.

File: src/coverlet.collector/DataCollection/CoverletCoverageCollector.cs (hunk 4 above)
Comment (daveMueller) [reply src/coverlet.collector/DataCollection/CoverletCoverageCollector.cs]: Done.

File: src/coverlet.console/Program.cs (hunk 5 above)
Comment (daveMueller) [reply src/coverlet.console/Program.cs]: Done.

File: src/coverlet.collector/DataCollection/CoverageManager.cs [hunk 6]
```diff
@@ -23,28 +23,28 @@ internal class CoverageManager
         public IReporter[] Reporters { get; }
 
         public CoverageManager(CoverletSettings settings, TestPlatformEqtTrace eqtTrace, TestPlatformLogger logger, ICoverageWrapper coverageWrapper,
-                               IInstrumentationHelper instrumentationHelper, IFileSystem fileSystem, ISourceRootTranslator sourceRootTranslator)
+            IInstrumentationHelper instrumentationHelper, IFileSystem fileSystem, ISourceRootTranslator sourceRootTranslator)
```
Comment (MarcoRossignoli) [inline src/coverlet.collector/DataCollection/CoverageManager.cs:26]: Sorry I wasn't clear, I don't want any update on this file, revert completely to master using command like `git checkout master src/coverlet.collector/DataCollection/CoverageManager.cs `

File: src/coverlet.console/Program.cs [hunk 7]
```diff
@@ -28,13 +28,13 @@ static int Main(string[] args)
             serviceCollection.AddTransient<ILogger, ConsoleLogger>();
             // We need to keep singleton/static semantics
             serviceCollection.AddSingleton<IInstrumentationHelper, InstrumentationHelper>();
-            serviceCollection.AddSingleton<ISourceRootTranslator, SourceRootTranslator>(serviceProvider => new SourceRootTranslator(serviceProvider.GetRequiredService<ILogger>(), serviceProvider.GetRequiredService<IFileSystem>()));
+            serviceCollection.AddSingleton<ISourceRootTranslator, SourceRootTranslator>(provider => new SourceRootTranslator(provider.GetRequiredService<ILogger>(), provider.GetRequiredService<IFileSystem>()));
 
-            DependencyInjection.Set(serviceCollection.BuildServiceProvider());
+            var serviceProvider = serviceCollection.BuildServiceProvider();
```
Comment (MarcoRossignoli) [inline src/coverlet.console/Program.cs:33]: please avoid `var` here

**Comment by MarcoRossignoli** [comment]
You can resolve conflicts taking master files that project is already without statics.

File: src/coverlet.collector/DataCollection/CoverageManager.cs (hunk 6 above)
Comment (daveMueller) [reply src/coverlet.collector/DataCollection/CoverageManager.cs]: Done

File: src/coverlet.console/Program.cs (hunk 7 above)
Comment (daveMueller) [reply src/coverlet.console/Program.cs]: Done

File: src/coverlet.msbuild.tasks/BaseTask.cs [hunk 8]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+using Microsoft.Build.Utilities;
+
+namespace Coverlet.MSbuild.Tasks
+{
+    public abstract class BaseTask : Task
+    {
+        protected static IServiceProvider ServiceProvider { get; set; }
```
Comment (MarcoRossignoli) [inline src/coverlet.msbuild.tasks/BaseTask.cs:8]: @tonerdo take a look here, now to keep the semantics on statics we have a base class for tasks where we set "shared" static state for tasks, this is the only place where we need to have statics.

**Comment by MarcoRossignoli** [comment]
@tonerdo there is an issue with pipelines doesn't trigger anymore.

**Comment by daveMueller** [comment]
Any news on the problem that the pipline doesn't trigger anymore?

**Pull request merged**